plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -324,29 +325,108 @@ public void testExplainAnalyzeWithDeleteWithSubquery()
+                        {"integer '3'"},
+                        {"bigint '4'"},
+                        {"decimal '3.14'"},
+                        {"decimal '1234567890.123456789'"},
+                        {"'test'"},
+                        {"char 'test'"},
+                        {"objectid('6216f0c6c432d45190f25e7c')"},
+                        {"date '1970-01-01'"},
+                        {"time '00:00:00.000'"},
+                        {"timestamp '1970-01-01 00:00:00.000'"},
+                        {"timestamp '1970-01-01 00:00:00.000 UTC'"},
+                },
+                comparisonOperator());
+    }
+
+    @Test
+    public void testPredicatePushdownCharWithPaddedSpace()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_predicate_pushdown_char_with_padded_space",
+                "(k, v) AS VALUES" +
+                        "   (-1, CAST(NULL AS char(3))), " +
+                        "   (0, CAST('' AS char(3)))," +
+                        "   (1, CAST(' ' AS char(3))), " +
+                        "   (2, CAST('  ' AS char(3))), " +
+                        "   (3, CAST('   ' AS char(3)))," +
+                        "   (4, CAST('x' AS char(3)))," +
+                        "   (5, CAST('x ' AS char(3)))," +
+                        "   (6, CAST('x  ' AS char(3)))," +
+                        "   (7, CAST('\0' AS char(3)))," +
+                        "   (8, CAST('\0 ' AS char(3)))," +
+                        "   (9, CAST('\0  ' AS char(3)))")) {
+            assertThat(query("SELECT k, v FROM " + table.getName() + " WHERE v = ''"))
+                    // The value is included because both sides of the comparison are coerced to char(3)
+                    .matches("VALUES (0, char ''), (1, char ' '), (2, char '  '), (3, char '   ')")
+                    .isFullyPushedDown();
+            assertThat(query("SELECT k, v FROM " + table.getName() + " WHERE v = 'x '"))
+                    // The value is included because both sides of the comparison are coerced to char(3)
+                    .matches("VALUES (4, char 'x'), (5, char 'x '), (6, char 'x  ')")
+                    .isFullyPushedDown();
+            assertThat(query("SELECT k, v FROM " + table.getName() + " WHERE v = '\0  '"))
+                    // The value is included because both sides of the comparison are coerced to char(3)
+                    .matches("VALUES (7, char '\0'), (8, char '\0 '), (9, char '\0  ')")
+                    .isFullyPushedDown();
+        }
+    }
+
+    @Test(dataProvider = "comparisonOperator")
+    public void testHighPrecisionDecimalPredicate(String operator)
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_high_precision_decimal_predicate",
+                "(col DECIMAL(34, 0))",
+                Arrays.asList("decimal '3141592653589793238462643383279502'", null))) {
```

Comment:
Creating a new table per operator is redundant because the table definition and row don't change. I would avoid using `dataProvider` and inline them.